backend/src/main/java/de/bund/digitalservice/ris/caselaw/adapter/transformer/DecisionTransformer.java

```diff
@@ -198,23 +198,48 @@ public static DecisionDTO transformToDTO(
   }
 
   private static void addSource(
-      DecisionDTO currentDto, DecisionDTOBuilder<?, ?> builder, CoreData coreData) {
-    if (coreData.source() != null) {
+      DecisionDTO currentDto, DecisionDTOBuilder<?, ?> builder, DocumentationUnit decision) {
+    if (decision.coreData().source() != null) {
       List<SourceDTO> existingSources =
           currentDto.getSource() != null
               ? new ArrayList<>(currentDto.getSource())
               : new ArrayList<>();
-      Integer rank = existingSources.size() + 1;
 
+      // if the source reference is not included in reference lists (= has been deleted), remove the
```

Comment:
Just for the sake of readability of this condition, can we rewrite it like this?

```
// Check if the first existing source has a reference and remove if it has been deleted in domain object
if (!existingSources.isEmpty()) {
      SourceDTO firstSource = existingSources.getFirst();
      var reference = firstSource.getReference();

      if (reference != null && !decisionContainsReferenceWithId(decision, reference.getId())) {
        firstSource.setReference(null); // Otherwise the source can not be deleted
      }
}
```

Reply:
Ja, das ist viel besser - danke!